visidata/loaders/npy.py

```diff
@@ -11,44 +12,79 @@ def open_npz(vd, p):
     return NpzSheet(p.base_stem, source=p)
 
 vd.option('npy_allow_pickle', False, 'numpy allow unpickling objects (unsafe)')
+vd.option('npy_matrix_enumerate', False, 'enumerate matrix rows and columns')
 
 class NpySheet(Sheet):
+    _transpose: bool = False
+    _matrix_enumerate: bool = False
+
     def iterload(self):
         numpy = vd.importExternal('numpy')
         if not hasattr(self, 'npy'):
-            self.npy = numpy.load(str(self.source), encoding='bytes', **self.options.getall('npy_'))
+            self.npy = numpy.load(str(self.source), encoding='bytes', allow_pickle=bool(self.options.get('npy_allow_pickle')))
         self.reloadCols()
-        yield from Progress(self.npy, total=len(self.npy))
+        if self._transpose:
+            source = self.npy[:,None]
+        else:
+            source = self.npy
+
+        nrows = len(self.npy)
+
+        if self._matrix_enumerate:
+            source = list(list((chain((i,), row))) for i, row in enumerate(source))
+
+        yield from Progress(source, nrows)
+
 
     def reloadCols(self):
+        self._matrix_enumerate = bool(self.options.get('npy_matrix_enumerate'))
+
         self.columns = []
-        for i, (name, fmt, *shape) in enumerate(self.npy.dtype.descr):
-            if not name:
-                continue
-            if shape:
-                t = anytype
-            elif 'M' in fmt:
-                self.addColumn(Column(name, type=date, getter=lambda c,r,i=i: str(r[i])))
-                continue
-            elif 'i' in fmt or 'u' in fmt:
-                t = int
-            elif 'f' in fmt:
-                t = float
+        if len(self.npy.shape)==1:
+            self._transpose = not bool(self.npy.dtype.names)
```

Comment:
Do we need to save off this `_transpose` as a member variable?  Can we make it a member function or property instead?